src/sonic-yang-models/yang-models/sonic-copp.yang

```diff
@@ -0,0 +1,243 @@
+						policer.  This value represents the depth of the token
+						bucket.";
+
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "cbs can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "cbs must be greater than or equal to cir";
+					}
+				}
+
+				leaf pir {
+					type uint64;
+					description
+						"Peak information rate for the dual-rate token bucket
+						policer.  This value represents the rate at which tokens
+						are added to the secondary bucket.";
+
+					must "((current()/../cir) and (current()/../cir > 0))" {
+						error-message "pir can't be configured without cir.";
+					}
+					must "(current() >= current()/../cir)" {
+						error-message "pir must be greater than or equal to cir";
+					}
+				}
+
+				leaf pbs {
+					type uint64;
+					description
+						"Excess burst size for the dual-rate token bucket policer.
+						This value represents the depth of the secondary bucket.";
+
+					must "((current()/../pir) and (current()/../pir > 0))" {
+						error-message "pbs can't be configured without pir.";
+					}
+					must "(current() >= current()/../pir)" {
+						error-message "pbs must be greater than or equal to pir";
+					}
+					must "((not(current()/../cbs)) or (current() >= current()/../cbs))" {
+						error-message "pbs must be greater than or equal to cbs";
+					}
+				}
+				
+				leaf green_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for green_action";
+						}
+					}
+					description
+						"Green action";
+				}
+				
+				leaf yellow_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for yellow_action";
+						}
+					}
+					description
+						"Yellow action";
+				}
+
+				leaf red_action {
+					type string {
+						pattern "(drop)|(forward)|(copy)|(copy_cancel)|(trap)|(log)|(deny)|(transit)" {
+							error-message "Invalid value passed for red_action";
+						}
+					}
+					description
+						"Red action";
+				}
+				
+				leaf genetlink_name {
+					type string {
+						pattern "(psample)" {
+							error-message "Invalid value passed for genetlink_name";
+						}
+					}
+					description
+						"genetlink name";
+				}
+				
+				leaf genetlink_mcgrp_name {
+					type string {
+						pattern "(packets)" {
+							error-message "Invalid value passed for genetlink_mcgrp_name";
+						}
+					}
+					description
+						"genetlink multicast group name";
+				}
+			}
+			/* end of list COPP_GROUP_LIST */
+		}
+		/* end of container COPP_GROUP */
+
+		container COPP_TRAP {
+			list COPP_TRAP_LIST {
+				key name;
+
+				leaf name {
+					type string;
+					description
+						"Control plane policing trap name";
+				}
+
+				leaf trap_ids {
+					type string;
```

Comment:
should this be enum?

Reply:
Response [here](https://github.com/Azure/sonic-buildimage/pull/7199/files#r614752199)